modules/improvedigitalBidAdapter.js

```diff
@@ -728,3 +734,61 @@ export function ImproveDigitalAdServerJSClient(endPoint) {
     return outputObject;
   };
 }
+
+export function RAZR() {
+  this.bids = {};
+
+  this.addBidData = data => {
+    const {bid} = data || {};
+
+    if (this.isMassBid(bid)) {
+      this.bids[bid.requestId] = {
+        ...data,
+        adm: bid.ad
+      };
+
+      bid.ad = `<script>window.top.postMessage({razrBidId: "${bid.requestId}"}, "*");\x3c/script>`;
+      this.addListenerOnce();
+    }
+  };
+
+  this.isMassBid = bid => {
+    return bid && /[rm]a[zs][rs]:\\?\/\\?\//.test(bid.ad);
+  };
+
+  this.render = (bidId, event) => {
+    const ns = window.razr = window.razr || {};
```

Comment:
* should be window.razrConfig (as in window.massConfig)
* This reminded me that we should also add a new adapter param for RAZR config so that publishers don't need to put a separate JS tag with window.razr on the page. I propose param name `rendererConfig` and will have the same structure as massConfig/razrConfig. https://docs.prebid.org/dev-docs/bidders/improvedigital.html

Reply:
This is not used for configuration, but to pass data between the adapter and the renderer. We can look for configuration under "rendererConfig", coming from the adapter config (this requires an update to renderer.js).

Reply:
Let's discuss this with Reda when you are back from holiday. Basically, for any new publisher adopting Razr we don't want to confuse them by asking them to add a new `window.razrConfig` tag on the page but rather we only ask them to update Improve Digital Prebid adapter + add Improve Prebid param renderConfig with the value being actually the same object as would have been `window.razrConfig`. We need to discuss also how to deal with existing MASS publishers and what would be the strategy for switching back to MASS if it's ever needed. Please set up a call on your return.